### PR TITLE
Adds Sentry.io client and server monitoring

### PR DESCRIPTION
**Pull request by jordanh**

### CODE REVIEW

**Comment by jordanh** [comment]
@mattkrick this is ready for your 👁 👁

File: src/universal/components/Auth0ShowLock/Auth0ShowLock.js [hunk 1]
```diff
@@ -27,9 +27,9 @@ export function showLock(dispatch) {
   // eslint-disable-next-line global-require
   const Auth0Lock = require('auth0-lock');
   let clientOptions = defaultClientOptions;
-  if (__PRODUCTION__) {
+  if (typeof __PRODUCTION__ !== 'undefined' && __PRODUCTION__) {
```
Comment (mattkrick) [inline src/universal/components/Auth0ShowLock/Auth0ShowLock.js]: `__PRDUCTION__` is guaranteed if this is built with webpack, why is this required?

File: src/server/Html.js [hunk 2]
```diff
@@ -27,7 +27,12 @@ export default function Html({store, entries, StyleSheetServer, renderProps}) {
     );
   });
   const dehydratedStyles = `window.__APHRODITE__ = ${JSON.stringify(css.renderedClassNames)}`;
-  const auth0ClientOptions = `window.__AUTH0__ = ${JSON.stringify(auth0)}`;
+  const clientOptions = `
+    window.__ACTION__ = {
+      auth0: ${JSON.stringify(auth0)},
+      sentry: ${JSON.stringify(process.env.SENTRY_DSN_PUBLIC)}
```
Comment (mattkrick) [inline src/server/Html.js]: could we shift this to using `webpack.EnvironmentPlugin`?

File: src/client/client.js [hunk 3]
```diff
@@ -25,8 +25,17 @@ const initialState = {};
      * During the production client bundle build, the server will need to be
      * stopped.
      */
-    // eslint-disable-next-line no-underscore-dangle
+    /* eslint-disable no-underscore-dangle */
     StyleSheet.rehydrate(window.__APHRODITE__);
+    if (window.__ACTION__.sentry) {
```
Comment (mattkrick) [inline src/client/client.js]: remove the conditional, since we want it to fail loudly.

File: src/universal/redux/authDuck.js [hunk 4]
```diff
@@ -43,6 +43,16 @@ export function setAuthToken(authToken, metaProfile = {}) {
     throw new Error(`unable to decode jwt: ${e}`);
   }
   cashay.create({httpTransport: new ActionHTTPTransport(authToken)});
+  if (typeof __PRODUCTION__ !== 'undefined' && __PRODUCTION__) {
```
Comment (mattkrick) [inline src/universal/redux/authDuck.js]: genuinely curious why the type check is required!

File: src/universal/components/Auth0ShowLock/Auth0ShowLock.js (hunk 1 above)
Comment (jordanh) [inline src/universal/components/Auth0ShowLock/Auth0ShowLock.js]: This one I can get rid of because it's not in the `ava` unit test path.

File: src/server/Html.js (hunk 2 above)
Comment (jordanh) [inline src/server/Html.js]: I don't think so. Not without shifting our build process to our server infrastructure. Webpack's `EnvironmentPlugin` will let us set these variables at compile time, but we want them at run time, inherited from the server's environment. I think this is a reasonable way to proceed for now...

File: src/client/client.js (hunk 3 above)
Comment (jordanh) [inline src/client/client.js]: ok!

File: src/server/Html.js (hunk 2 above)
Comment (jordanh) [inline src/server/Html.js]: That's right, we'll have two separate DSNs. One for staging and one for production.
